share/actionsmap.yml

```diff
@@ -1396,79 +1396,85 @@ firewall:
                     full: --raw
                     help: Return the complete YAML dict
                     action: store_true
-                -i:
-                    full: --by-ip-version
-                    help: List rules by IP version
-                    action: store_true
+                protocol:
+                    help: "If not raw, protocol type to list (tcp/udp)"
+                    choices:
+                        - tcp
+                        - udp
+                    nargs: "?"
                 -f:
-                    full: --list-forwarded
-                    help: List forwarded ports with UPnP
+                    full: --forwarded
+                    help: If not raw, list UPnP forwarded ports instead of open ports
                     action: store_true
 
-        ### firewall_allow()
-        allow:
+        ### firewall_open()
+        open:
             action_help: Allow connections on a port
-            api: PUT /firewall/<protocol>/allow/<port>
+            api: PUT /firewall/<protocol>/open/<port>
             arguments:
-                protocol:
-                    help: "Protocol type to allow (TCP/UDP/Both)"
-                    choices:
-                        - TCP
-                        - UDP
-                        - Both
-                    default: TCP
                 port:
                     help: Port or range of ports to open
                     extra:
                         pattern: &pattern_port_or_range
                             - !!str ((^|(?!\A):)([0-9]{1,4}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5])){1,2}?$
                             - "pattern_port_or_range"
-                -4:
-                    full: --ipv4-only
-                    help: Only add a rule for IPv4 connections
-                    action: store_true
-                -6:
-                    full: --ipv6-only
-                    help: Only add a rule for IPv6 connections
-                    action: store_true
-                --no-upnp:
-                    help: Do not add forwarding of this port with UPnP
+                protocol:
+                    help: "Protocol type (tcp/udp)"
+                    choices:
+                        - tcp
+                        - udp
+                    default: tcp
+                comment:
+                    help: A reason for the port to be open (like the app's name)
+                    default: ""
```

Comment:
I would switch these to keyword args (`--protocol / -p` and `--comment / -c`) othewise it's a pain to have them as positional and having to explicitly set "tcp" as protocol when you just want to define the comment

Reply:
Hmmm so if not passed, protocol should be just tcp ?